### PR TITLE
feat: change B2C Job Allowlist to admin config

### PR DESCRIPTION
**Pull request by cdeery**
This is a followup to https://github.com/openedx/taxonomy-connector/pull/156 with a non-config based solution.
The B2CJobAllowlist uses the Admin interface to manage the jobs that should be on the allowlist.

**Merge checklist:**
- [x] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `make upgrade && make requirements` have been run to regenerate requirements
- [x] `./manage.py makemigrations` has been run
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
- [x] [Version](https://github.com/openedx/taxonomy-connector/blob/master/taxonomy/__init__.py) bumped
- [x] [Changelog](https://github.com/openedx/taxonomy-connector/blob/master/CHANGELOG.rst) record added

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/openedx/taxonomy-connector/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [ ] After versioned build finishes in [GitHub Actions](https://github.com/openedx/taxonomy-connector/actions), verify version has been pushed to [PyPI](https://pypi.org/project/taxonomy-connector/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [course-discovery](https://github.com/openedx/course-discovery) to upgrade dependencies (including taxonomy-connector)
    - This **must** be done after the version is visible in PyPi as `make upgrade` in course-discovery will look for the latest version in PyPi.

### CODE REVIEW

File: taxonomy/migrations/0033_auto_20230614_1246.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+            options={
+                'verbose_name': 'B2C Job Allow List entry',
+                'verbose_name_plural': 'B2C Job Allow List entries',
+                'ordering': ('external_id',),
+            },
+        ),
+        migrations.AlterField(
+            model_name='courserunxblockskillstracker',
+            name='course_run_key',
+            field=models.CharField(help_text='Course run key of the course under which all xblocks were tagged.', max_length=255, unique=True),
```
Comment (justinhynes) [inline taxonomy/migrations/0033_auto_20230614_1246.py:28]: Huh, strange. It looks like there was possibly a missing migration that got combined with your changes. Do we want to include this?

File: taxonomy/migrations/0033_auto_20230614_1246.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
+# Generated by Django 3.2.18 on 2023-06-14 12:46
```
Comment (justinhynes) [inline taxonomy/migrations/0033_auto_20230614_1246.py:1]: Can we update the name of the auto-generated migration to represent the database changes? Maybe something like... `0033_add_b2c_job_allowlist`?

File: taxonomy/models.py [hunk 3]
```diff
@@ -1042,3 +1042,43 @@ class Meta:
...
+
+    .. no_pii:
+    """
+    external_id = models.CharField(
+        max_length=255,
+        blank=False,
+        null=False,
+        unique=True,
+        help_text=_(
+            'The external identifier for the job received from API.'
```
Comment (justinhynes) [inline taxonomy/models.py:1059]: nit: Maybe we should add a bit more detail like `EMSI API`?

Comment (justinhynes) [inline taxonomy/models.py:1059]: I had this half-baked assumption that this model field would be a one-to-one relationship to a Job. Any benefit to not having a relationship to the other model?

Comment (cdeery) [reply taxonomy/models.py:1059]: I was going to ask you about that! I think I was being to literal about mapping the config. I can bake that more fully. Is there any problem with an update making a job go away, and a mapping to a job blowing up?

File: taxonomy/migrations/0033_auto_20230614_1246.py (hunk 2 above)
Comment (cdeery) [reply taxonomy/migrations/0033_auto_20230614_1246.py]: Changed. it's now 0033_b2c_allowlist.py

File: taxonomy/models.py (hunk 3 above)
Comment (cdeery) [reply taxonomy/models.py]: Done

Comment (cdeery) [reply taxonomy/models.py]: Done. It makes more sense now.

**Review by justinhynes — APPROVED** [review]

File: taxonomy/models.py [hunk 4]
```diff
@@ -1042,3 +1042,42 @@ class Meta:
...
+        return '<External Id = "{}" Job ="{}">'.format(self.job.external_id, self.job)
+
+    class Meta:
+        """
+        Meta configuration for B2C Job Allow List model.
+        """
+
+        app_label = 'taxonomy'
+        verbose_name = 'B2C Job Allow List entry'
+        verbose_name_plural = 'B2C Job Allow List entries'
```
Comment (sameenfatima78) [inline taxonomy/models.py:1083]: Can we add test cases for this model and methods? This will help fix the coverage failure too.

Comment (cdeery) [reply taxonomy/models.py:1083]: Done, and yes, it fixed the code coverage problem. Thanks

**Comment by cdeery** [comment]
@sameenfatima78 Are there any other changes or suggestions?

File: taxonomy/migrations/0033_auto_20230614_1246.py (hunk 1 above)
Comment (cdeery) [reply taxonomy/migrations/0033_auto_20230614_1246.py]: @sameenfatima78 Is this ok to include?

**Review by sameenfatima78 — APPROVED** [review]
Good to merge.

**Comment by sameenfatima78** [comment]
> @sameenfatima78 Are there any other changes or suggestions?

Nope, it looks good. Should I merge it?

**Pull request merged**